### PR TITLE
Dilpreet | Test | Remove implemented combine mobile no

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
https://app.clickup.com/t/2p1fd62

### CODE REVIEW

**Comment by dilpreetsinghofficial** [comment]
Reviewed

**Review by the-prateek-mishra — APPROVED** [review]

**Comment by dilpreetsinghofficial** [comment]
Reviewed

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 1]
```diff
@@ -7892,12 +7859,72 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
...
-        this.generalService.focusInCountrySearch(element);
+*This will use for  fetch mobile number
+*
+* @memberof ProformaInvoiceComponent
+*/
+    public onlyPhoneNumber() {
+        const input = document.getElementById('init-contact-proforma');
+        this.intl = new window['intlTelInput'](input, {
+            nationalMode: false,
+            utilsScript: 'https://cdnjs.cloudflare.com/ajax/libs/intl-tel-input/17.0.17/js/utils.js',
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]: create url as constant

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 2]
```diff
@@ -7892,12 +7859,72 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
...
-     * @param {*} element
-     * @memberof ProformaInvoiceComponent
-     */
-    public focusInCallingCodeSearch(element: any): void {
-        this.generalService.focusInCountrySearch(element);
+*This will use for  fetch mobile number
+*
+* @memberof ProformaInvoiceComponent
+*/
+    public onlyPhoneNumber() {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]: add return type

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 3]
```diff
@@ -7892,12 +7859,72 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
...
+            },
+        });
+        input.addEventListener('blur', () => {
+            if (!this.intl?.isValidNumber()) {
+                this.showMobileNumberError = true;
+            } else {
+                this.showMobileNumberError = false;
+            }
+        });
+        input.addEventListener('countrychange', () => {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]: remove blank event listener

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts [hunk 4]
```diff
@@ -537,6 +527,30 @@ export class AccountAddNewDetailsComponent implements OnInit, OnChanges, AfterVi
         this.addAccountForm.reset();
     }
 
+    public isValidMobileNumber(ele: HTMLInputElement) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]: add return type and js doc

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts [hunk 5]
```diff
@@ -537,6 +527,30 @@ export class AccountAddNewDetailsComponent implements OnInit, OnChanges, AfterVi
...
+        if (ele.value) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]:
```suggestion
        if (element?.value) {
```

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts (hunk 4 above)
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]:
```suggestion
    public isValidMobileNumber(element: HTMLInputElement) {
```

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts [hunk 6]
```diff
@@ -537,6 +527,30 @@ export class AccountAddNewDetailsComponent implements OnInit, OnChanges, AfterVi
...
+            this.checkMobileNo(ele);
+        }
+    }
+
+    public checkMobileNo(ele) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]:
```suggestion
    public checkMobileNo(element) {
```

Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]: add return type and js doc

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts [hunk 7]
```diff
@@ -1235,30 +1252,73 @@ export class AccountAddNewDetailsComponent implements OnInit, OnChanges, AfterVi
...
     /**
-     * Will put focus in search field in calling code dropdown
+      *This will use for  fetch mobile number
      *
-     * @param {*} element
      * @memberof AccountAddNewDetailsComponent
      */
-    public focusInCallingCodeSearch(element: any): void {
-        this.generalService.focusInCountrySearch(element);
+    public onlyPhoneNumber() {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]: add return type

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 8]
```diff
@@ -651,6 +639,29 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
         this.addBlankGstForm();
     }
 
+    public isValidMobileNumber(ele: HTMLInputElement) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]:
```suggestion
    public isValidMobileNumber(element: HTMLInputElement) {
```

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 9]
```diff
@@ -651,6 +639,29 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
...
+        if (ele.value) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]:
```suggestion
        if (element?.value) {
```

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts (hunk 8 above)
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]: add return type and js doc

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 10]
```diff
@@ -651,6 +639,29 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
...
+            this.checkMobileNo(ele);
+        }
+    }
+
+    public checkMobileNo(ele) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]: add return type and js doc

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 11]
```diff
@@ -1790,29 +1824,72 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
...
-     * @param {*} element
-     * @memberof AccountUpdateNewDetailsComponent
-     */
-    public focusInCallingCodeSearch(element: any): void {
-        this.generalService.focusInCountrySearch(element);
+    *This will use for  fetch mobile number
+   *
+   * @memberof AccountUpdateNewDetailsComponent
+   */
+    public onlyPhoneNumber() {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]: add return type and js doc

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts [hunk 12]
```diff
@@ -583,8 +573,9 @@ export class AccountAddNewDetailsComponent implements OnInit, OnChanges, AfterVi
             delete accountRequest['addresses'];
         }
 
-        let mobileNo = this.addAccountForm?.get('mobileNo')?.value?.e164Number;
+        let mobileNo = this.intl.getNumber();
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]:
```suggestion
        let mobileNo = this.intl?.getNumber();
```

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 13]
```diff
@@ -713,8 +704,10 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
             this.addAccountForm.get('currency')?.patchValue(this.selectedCurrency, { onlySelf: true });
             accountRequest.currency = this.selectedCurrency;
         }
-        let mobileNo = this.addAccountForm?.get('mobileNo')?.value?.e164Number;
+
+        let mobileNo = this.intl.getNumber();
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]:
```suggestion
        let mobileNo = this.intl?.getNumber();
```

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**